models.py

```diff
@@ -232,9 +232,37 @@ def location_network(rnn_state, sampling, std=params['std'],
     # core network
     # TODO: parameterize rnn_cell, i.e. implement split cell from paper
     with tf.variable_scope('core_network', reuse=tf.AUTO_REUSE):
-        rnn_cell = tf.nn.rnn_cell.LSTMCell(params['core_size'])
-        state = rnn_cell.zero_state(batch_size, tf.float32)
-
+        if params['core_type'] == 'LSTM':
+            rnn_cell = tf.nn.rnn_cell.LSTMCell(params['core_size'])
+            state = rnn_cell.zero_state(batch_size, tf.float32)
+
+        else:
+            def rnn_cell(glimpse, state): 
+                
+                dense_state = tf.layers.dense(
+                    inputs=state,
+                    units=params['core_size'],
+                    activation=None)
+            
+                dense_glimpse = tf.layers.dense(
+                    inputs=glimpse,
+                    units=params['glimpse_out_size'],
```

Comment:
For the `tf.add` below, this needs to have the same number of units as `dense_state`.  I would use `params['core_size']` for both :-).